### PR TITLE
Cloud Native Chapter 4

### PR DESCRIPTION
**Pull request by ardalis**
## Summary

Chapter 4 files - toc and images are in Ch1 PR.

### CODE REVIEW

**Comment by mairaw** [comment]
can you fix those warnings about en-us in the docs URLs while I review the other PRs?

File: docs/architecture/cloud-native/communication-considerations.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Communication Considerations
+description: Architecting Cloud Native .NET Apps for Azure | Communication Considerations
+ms.date: 06/30/2019
+---
+# Communication Considerations
+
+Take a close look at the apps in your portfolio. How many of them justify a cloud native architecture? All of them? Perhaps some?
+
+Applying a cost/benefit trade-off, many would not warrant the hefty price tag required to be cloud native. A cloud native-based approach would greatly exceed the business value of the application � like driving a large 18-wheeler semi-truck to the local 7-11 store to buy milk � when your simple Honda sedan would work just fine.
```
Comment (mairaw) [inline docs/architecture/cloud-native/communication-considerations.md]: can you get rid of these � here too?

Comment (ardalis) [reply docs/architecture/cloud-native/communication-considerations.md]: done

File: docs/architecture/cloud-native/communication-considerations.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+- The ability to immediately release features without redeploying the entire system
+
+- Applications developed by heterogenous teams with expertise in different technology stacks
+
+- Applications with components that must scale independently
+
+While we'd all like to maximize our developer excitement level building brand new (i.e., Greenfield) systems, often we're charged with modernizing legacy (i.e., Brownfield) workloads that are critical to the business.
+
+The free Microsoft eBook [Modernize existing .NET applications with Azure cloud and Windows Containers](https://dotnet.microsoft.com/download/thank-you/modernizing-existing-net-apps-ebook) provides a wealth of guidance for migrating on-premises workloads into cloud. Figure 1-11, show below, shows that there isn't a single, one-size-fits-all strategy for migrating legacy applications to the cloud.
```
Comment (EugeneKramer) [inline docs/architecture/cloud-native/communication-considerations.md]:
```suggestion
The free Microsoft eBook [Modernize existing .NET applications with Azure cloud and Windows Containers](https://dotnet.microsoft.com/download/thank-you/modernizing-existing-net-apps-ebook) provides a wealth of guidance for migrating on-premises workloads into cloud. Figure 1-11, shown below, shows that there isn't a single, one-size-fits-all strategy for migrating legacy applications to the cloud.
```

File: docs/architecture/cloud-native/front-end-communication.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+**Figure 4-7**. Ocelot Features
+
+Each Ocelot gateway instance includes a simple JSON configuration file that specifies the upstream and downstream addresses and configurable features as shown in Figure 4-8:
+
+![Basic Ocelot implementation](media/basic-ocelot-implementation.png)
+**Figure 4-8**. Basic Ocelot implementation
+
+In Figure 4-8, the client sends an HTTP request to the Ocelot gateway. Once received, Ocelot manipulates the HttpRequest object into a state specified by its configuration. At the end of pipeline, Ocelot creates a new HttpRequestMessage that is passed to the downstream service. In reverse, Ocelot receives the HTTP response and sends it back to the client.
+
+Ocelot is extensible and can support many modern platforms, including Azure Kubernetes Services and Service Fabric, as well as integration with open-source packages like Consul, GrapQL, Netflix’s Eureka, web sockets, and SignalR.
```
Comment (mairaw) [inline docs/architecture/cloud-native/front-end-communication.md]: Did you mean GraphQL here?

```suggestion
Ocelot is extensible and can support many modern platforms, including Azure Kubernetes Services and Service Fabric, as well as integration with open-source packages like Consul, GraphQL, Netflix’s Eureka, web sockets, and SignalR.
```

**Comment by mairaw** [comment]
This Chapter is reviewed. Please take a look at my edits.

**Review by mairaw — APPROVED** [review]
Chapter reviewed. Please confirm it's good to merge @ardalis.

**Review by ardalis — COMMENTED** [review]
LGTM

**Comment by ardalis** [comment]
:shipit:

**Pull request merged**